distutils/_msvccompiler.py

```diff
@@ -527,8 +526,12 @@ def _fallback_spawn(self, cmd, env):
             return
         warnings.warn(
             "Fallback spawn triggered. Please update distutils monkeypatch.")
-        with unittest.mock.patch('os.environ', env):
+        old_path = os.getenv("PATH")
```

Comment:
Is PATH the only variable that needs to be patched? Can you implement a contextmanager that has the equivalent simplicity of `unittest.mock.patch`?

Reply:
I did some searching and found [this article](https://adamj.eu/tech/2020/10/13/how-to-mock-environment-variables-with-pythons-unittest/), which suggests using mock.patch.dict on os.environ will have the intended effect.